pkgs/development/python-modules/mlx/default.nix

```diff
@@ -65,28 +72,38 @@ buildPythonPackage rec {
     ];
   };
 
-  nativeBuildInputs = [
+  build-system = [
     cmake
+    nanobind
+    setuptools
+  ];
+  dontUseCmakeConfigure = true;
+
+  nativeBuildInputs = [
     pybind11
     xcbuild
     zsh
     gguf-tools
     nlohmann_json
-    setuptools
   ];
 
   buildInputs = [
     blas
     lapack
+    fmt
+  ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
   ];
 
-  meta = with lib; {
+  meta = {
     homepage = "https://github.com/ml-explore/mlx";
     description = "Array framework for Apple silicon";
     changelog = "https://github.com/ml-explore/mlx/releases/tag/v${version}";
-    license = licenses.mit;
+    license = lib.licenses.mit;
     platforms = [ "aarch64-darwin" ];
-    maintainers = with maintainers; [
```

Comment:
I'd decouple these semantic changes to a separate commit.

Reply:
Sure, I gathered them in a "cleanup" commit.